osmaxx-py/osmaxx/contrib/auth/frontend_permissions.py

```diff
@@ -45,3 +46,26 @@ class FrontendAccessRequiredMixin(object):
     @method_decorator(frontend_access_required)
     def dispatch(self, *args, **kwargs):
         return super(FrontendAccessRequiredMixin, self).dispatch(*args, **kwargs)
+
+
+class AuthenticatedAndAccessPermission(permissions.BasePermission):
+    """
+    Allows access only to authenticated users with frontend permissions.
+    """
+
+    def has_permission(self, request, view):
+        return request.user.is_authenticated() and _may_user_access_osmaxx_frontend(request.user)
+
+
+class HasBBoxAccessPermission(permissions.BasePermission):
+    message = 'Accessing this bounding box is not allowed.'
+
+    def has_object_permission(self, request, view, obj):
+        return obj.excerpt.is_public or obj.excerpt.owner == request.user
```

Comment:
On other thoughts, permissions should not be a concern of the model itself, since it can be viewed as orthogonal to the model itself. I'm therefore placing it in a method in permissions, agree?